bands/models.py

```diff
@@ -59,7 +59,10 @@ def email(self):
 
     @property
     def normalized_group_name(self):
-        return ''.join(e for e in self.group.name.strip().lower() if e.isalnum() or e == ' ').replace(' ', '_')
+        # Remove all special characters, then remove all extra whitespace, then replace spaces with underscores
+        return ' '.join(
+            ''.join(e for e in self.group.name.strip().lower() if e.isalnum() or e == ' ').split()
+        ).replace(' ', '_')
```

Comment:
Yeah, it certainly looks weird, but I didn't see a better way to do everything I wanted, including:
1. remove extra whitespace but not ALL whitespace, collapsing multiple spaces into one, etc
2. remove all special characters _except_ whitespace, because next we're going to...
3. convert whitespace to underscores